code/platform_win32/win32_4ed.cpp

```diff
@@ -780,12 +780,12 @@ win32_keycode_init(void){
     keycode_lookup_table[VK_NUMPAD9] = KeyCode_NumPad9;
     
     for (i32 i = 0xDF; i < 0xFF; i += 1){
-        keycode_lookup_table[i] = KeyCode_Ex0 + 1;
+        keycode_lookup_table[i] = KeyCode_Ex0 + i - 0xDF;
```

Comment:
This results in `0xff - 0xdf = 32` extension codes but the keycode enum in: [4coder_event_codes.h](https://github.com/4coder-community/4cc/blob/master/code/custom/generated/4coder_event_codes.h) only does 0 through 29 so this will overflow that enum?
We should probably fix that, although we'd have to figure out what generates that file

Reply:
Nice catch. I didn't check the loop bounds. It seems that `4ed_generate_keycodes.cpp` is what generates those. I'll have a look when I've got sometime. The linux code expects the max value to be Ex29 so, we might just need to loop for less values.

Reply:
I changed the loop to do the same thing as the file generating the keycode enums. It loops between 0 (included) and 30 (not included) indicating that the intent was to create 30 Ex values.
I also updated the keycode enum generator to have a comment specifying which file generated it, and also to generate proper indentation.
Note that the keycode enum generator doesn't run with the build system. It needs to be manually called when necessary.